x-pack/test/api_integration/apis/monitoring/data_stream.ts

```diff
@@ -22,11 +23,12 @@ export const getLifecycleMethods = (getService) => {
   };
 
   return {
-    async setup(archive) {
+    async setup(archive: string) {
       await esArchiver.load(archive, { useCreate: true });
     },
 
     async tearDown() {
+      await deleteDataStream('metricbeat-*');
```

Comment:
The fact that we have to delete the datastream instead of calling `esArchiver.unload(archive)` is because the monitoring mappings are already installed by elasticsearch and we don't need to set them up, thus the archiver don't have any reference to the template and can't automatically delete it.
We could however do it for the metricbeat archive for a complete cleanup of the assets.

This weirdness makes me think that we should maybe install an archived version of the mappings for `.monitoring-{product}-mb` to have a standardized usage of the archiver. we can discuss that in https://github.com/elastic/kibana/issues/119658

Reply:
I think I had the unload here running before the `deleteDataStream`. I'll put it back and see the behaviour. I don't remember if all the data got removed when I deleted the datastream.

Reply:
Apparently deleting a datastream also deletes its indexes https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-delete-data-stream.html, which makes sense, because I've run test test multiple times in a row and it never failed due to duplicate ids

Reply:
If we only delete the data stream I think we'll leave the backing index template/component templates of the ds

Reply:
True! The index template stays there if we don't unload the archive. I'll add that.